File: express/app/src/main/java/com/changsukuaidi/www/modules/login/launcher/LauncherRepository.java
package com.changsukuaidi.www.modules.login.launcher;

import com.changsukuaidi.www.bean.BaseJson;
import com.changsukuaidi.www.bean.CarBean;
import com.changsukuaidi.www.remote.ClientManager;
import com.changsukuaidi.www.remote.PunchClient;

import javax.inject.Inject;

import io.reactivex.Observable;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;


/**
 * @Describe
 * @Author thc
 * @Date 2017/12/6
 */

class LauncherRepository {
    private PunchClient mPunchClient;

    @Inject
    LauncherRepository(ClientManager clientManager) {
        mPunchClient = clientManager.providePunchClient();
    }

    Observable<BaseJson<CarBean>> getCarInfo() {
        return mPunchClient.getCarInfo()
                .subscribeOn(Schedulers.newThread())
                .unsubscribeOn(Schedulers.newThread())
                .observeOn(AndroidSchedulers.mainThread());
    }
}

File: WIFIDirect/app/src/main/java/com/thc/www/p2p/android/launcher/LauncherRepository.kt
package com.thc.www.p2p.android.launcher

import com.thc.www.p2p.bean.TokenBean
import com.thc.www.p2p.dao.TokenDao
import javax.inject.Inject

/**
 * @Describe
 * @Author thc
 * @Date  2018/5/7
 */
class LauncherRepository @Inject constructor(private val tokenDao: TokenDao) {
    fun getLoginInfo(): TokenBean {
        val tokenBeans: List<TokenBean> = tokenDao.getToken()
        return if (tokenBeans.isEmpty()) {
            TokenBean(0, "")
        } else {
            tokenBeans[0]
        }
    }
}